Common/cpp/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -230,23 +248,81 @@ void NativeReanimatedModule::unregisterEventHandler(
 }
 
+inline std::string int_to_hex(int val) {
+  std::stringstream ss;
+  ss << std::setfill('0') << std::setw(6) << std::hex << (val | 0);
+  return "#" + ss.str();
+}
+
+#ifdef RCT_NEW_ARCH_ENABLED
+
+std::string NativeReanimatedModule::obtainPropFromShadowNode(
+    jsi::Runtime &rt,
+    const std::string &propName,
+    const ShadowNode::Shared &shadowNode) {
+  auto newestCloneOfShadowNode =
+      uiManager_->getNewestCloneOfShadowNode(*shadowNode);
+  Props::Shared props = newestCloneOfShadowNode->getProps();
+  auto staticProps = std::static_pointer_cast<const ViewProps>(props);
+  auto layoutableShadowNode =
+      traitCast<LayoutableShadowNode const *>(newestCloneOfShadowNode.get());
+  const auto &frame = layoutableShadowNode->layoutMetrics_.frame;
+  std::string resultStr;
+  if (propName == "width") {
+    resultStr = std::to_string(frame.size.width);
+  } else if (propName == "height") {
+    resultStr = std::to_string(frame.size.width);
+  } else if (propName == "top") {
+    resultStr = std::to_string(frame.origin.y);
+  } else if (propName == "left") {
+    resultStr = std::to_string(frame.origin.x);
+  } else if (propName == "opacity") {
+    resultStr = std::to_string(staticProps->opacity);
+  } else if (propName == "zIndex") {
+    std::optional<int> zIndex = staticProps->zIndex;
+    if (zIndex) {
+      resultStr = std::to_string(*zIndex);
+    }
+  } else if (propName == "backgroundColor") {
+    resultStr = int_to_hex(*staticProps->backgroundColor);
+  }
+  return resultStr;
+}
+
+#endif
+
 jsi::Value NativeReanimatedModule::getViewProp(
     jsi::Runtime &rnRuntime,
     const jsi::Value &viewTag,
     const jsi::Value &propName,
+    const jsi::Value &shadowNodeWrapper,
     const jsi::Value &callback) {
-#ifdef RCT_NEW_ARCH_ENABLED
-  throw std::runtime_error(
-      "[Reanimated] getViewProp is not implemented on Fabric yet");
-#else
-  const int viewTagInt = viewTag.asNumber();
   const auto propNameStr = propName.asString(rnRuntime).utf8(rnRuntime);
   const auto funPtr = std::make_shared<jsi::Function>(
       callback.getObject(rnRuntime).asFunction(rnRuntime));
+#ifdef RCT_NEW_ARCH_ENABLED
+  const ShadowNode::Shared shadowNode =
+      shadowNodeFromValue(rnRuntime, shadowNodeWrapper);
+  uiScheduler_->scheduleOnUI([=]() {
+    jsi::Runtime &uiRuntime = uiWorkletRuntime_->getJSIRuntime();
+    std::string resultStr =
+        obtainPropFromShadowNode(uiRuntime, propNameStr, shadowNode);
+
+    jsScheduler_->scheduleOnJS([=](jsi::Runtime &rnRuntime) {
+      const auto resultValue =
+          jsi::String::createFromUtf8(rnRuntime, resultStr);
+      funPtr->call(rnRuntime, resultValue);
+    });
+  });
+
+  return jsi::Value::undefined();
+
+#else
+  const int viewTagInt = viewTag.asNumber();
 
   uiScheduler_->scheduleOnUI([=]() {
     jsi::Runtime &uiRuntime = uiWorkletRuntime_->getJSIRuntime();
-    const auto propNameValue =
+    const jsi::Value propNameValue =
```

Comment:
It looks that this is `jsi::String` by default, not `jsi::Value`

Reply:
This error appears otherwise:
<img width="427" alt="image" src="https://github.com/software-mansion/react-native-reanimated/assets/56199675/94653681-f593-4f4b-a2ed-65863aa3345f">